### PR TITLE
NuttX: migrate NuttX CMake build for WAMR

### PR DESCRIPTION
**Pull request by xuxin930**
Add NuttX CMake build support for platforms

### CODE REVIEW

**Comment by wenyongh** [comment]
@no1wudi Could you help have a look? Thanks a lot.

File: product-mini/platforms/nuttx/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+if(CONFIG_INTERPRETERS_WAMR_THREAD_MGR)
+  # include thread_mgr.cmake
+  set(WAMR_BUILD_THREAD_MGR 1)
+endif()
+
+if(CONFIG_INTERPRETERS_WAMR_LIB_PTHREAD)
+  # include lib_pthread.cmake
+  set(WAMR_BUILD_LIB_PTHREAD 1)
+endif()
```
Comment (no1wudi) [inline product-mini/platforms/nuttx/CMakeLists.txt:159]: Missing WAMR_BUILD_GC and WAMR_BUILD_STRINGREF

Comment (xuxin930) [reply product-mini/platforms/nuttx/CMakeLists.txt:184]: > Missing WAMR_BUILD_GC and WAMR_BUILD_STRINGREF

I added WAMR_BUILD_GC in the latest patchset, but I can't seem to find where to set WAMR_BUILD_STRINGREF

Comment (no1wudi) [reply product-mini/platforms/nuttx/CMakeLists.txt:184]: It's in `build-scripts/config_common.cmake:348`

Comment (xuxin930) [reply product-mini/platforms/nuttx/CMakeLists.txt:186]: > WAMR_BUILD_STRINGREF

😁 thx for you help, i added it to the latest patchset
```cmake
if(CONFIG_INTERPRETERS_WAMR_GC)
  set(WAMR_BUILD_GC 1)
  set(WAMR_BUILD_STRINGREF 1)
  set(WAMR_BUILD_REF_TYPES 1)
endif()
```

**Comment by no1wudi** [comment]
@xuxin930 Please trigger CI again

**Comment by xuxin930** [comment]
> @xuxin930 Please trigger CI again

It seems that there is something wrong with the environment settings of the arm MacOs.

**Comment by wenyongh** [comment]
> > @xuxin930 Please trigger CI again
> 
> It seems that there is something wrong with the environment settings of the arm MacOs.

It is a little strange that macos CI ran failed for this PR:
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/8502961539/job/23289867170
```bash
Run /usr/bin/env python3 -m pip install -r requirements.txt
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try brew install
    xyz, where xyz is the package you are trying to
    install.
```
While other PR ran well:
https://github.com/bytecodealliance/wasm-micro-runtime/actions/runs/8503551804/job/23289206815

**Comment by wenyongh** [comment]
@xuxin930 @no1wudi I submitted PR #3270 to fix the macos CI issue.

**Comment by no1wudi** [comment]
@xuxin930 Please rebase the main branch to fix the macOS CI.

**Pull request merged**